Sources/TuistKit/Commands/EnvKey/EnvKey.swift

```diff
@@ -232,10 +232,7 @@ extension EnvKey {
         return T(argument: envValueString)
     }
 
-    func envValue<T: ExpressibleByArgument>() -> [T] {
-        guard let envValueString else {
-            return []
-        }
-        return envValueString.split(separator: ",").compactMap { T(argument: String($0)) }
+    func envValue<T: ExpressibleByArgument>() -> [T]? {
+        return envValueString?.split(separator: ",").compactMap { T(argument: String($0)) }
```

Comment:
If we return an empty array when it should be nil, we would successfully unwrap the value [here](https://github.com/tuist/tuist/blob/main/Sources/TuistKit/Commands/EnvKey/ParseableArgument%2BExtension.swift#L150) and we would never use the default value specified in the command in the [else block](https://github.com/tuist/tuist/blob/main/Sources/TuistKit/Commands/EnvKey/ParseableArgument%2BExtension.swift#L157-L163).

Reply:
Good catch!